### PR TITLE
Adds service configuration rumor and encryption

### PR DESCRIPTION
**Pull request by adamhjk**
gif-keyboard-16468722929976415170

* Adds the ServiceConfig rumor
* Adds optional encryption to the wire
* Adds a "Wire" message type, used to wrap all messages with optional
  encryption

Signed-off-by: Adam Jacob <adam@chef.io>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: components/butterfly/src/service_config.rs [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        }
+    }
+
+    fn kind(&self) -> ProtoRumor_Type {
+        ProtoRumor_Type::ServiceConfig
+    }
+
+    fn id(&self) -> &str {
+        "config"
+        // self.get_member_id()
```
Comment (reset) [inline components/butterfly/src/service_config.rs]: Should this be removed? Do we intend to always have an ID of `"config"`?

File: components/core/src/crypto/keys/sym_key.rs [hunk 2]
```diff
@@ -28,6 +29,12 @@ use super::super::{SECRET_SYM_KEY_SUFFIX, SECRET_SYM_KEY_VERSION, hash};
 
 pub type SymKey = KeyPair<(), SymSecretKey>;
 
+impl fmt::Debug for SymKey {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "SymKey")
```
Comment (reset) [inline components/core/src/crypto/keys/sym_key.rs]: Should we include some sort of identifier here to differentiate in debug output which SymKey is being displayed?

**Comment by reset** [comment]
@adamhjk few comments inline but all looks good!

**Comment by reset** [comment]
gif-keyboard-17137461516503525047

File: components/butterfly/src/service_config.rs (hunk 1 above)
Comment (adamhjk) [inline components/butterfly/src/service_config.rs]: Yeah, we do. There is only one config rumor per service group.

**Comment by reset** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/171672438).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/171672438).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**